build_script.py

```diff
@@ -184,40 +185,60 @@ def main(args=sys.argv[1:]):
+        description=textwrap.dedent("""
+            Build, test, and install XCTest.
+
+            WARNING: This script is only meant to be used on Linux host
+            machines, and in general should not be invoked directly. The
+            recommended way to build and test XCTest is via the Swift build
+            script. See this project's README for details.
+
+            The Swift build script invokes this %(prog)s script to build,
+            test, and install this project on Linux. Assuming you are using a
+            Linux host machine, you may invoke it in the same way to build this
+            project directly. For example, If your install of Swift is located
+            at "/swift" and you wish to install XCTest into that same location,
+            here is a sample invocation of the build script:
+
+            $ %(prog)s \\
+                --swiftc="/swift/usr/bin/swiftc" \\
+                --build-dir="/tmp/XCTest_build" \\
+                --foundation-build-dir "/swift/usr/lib/swift/linux" \\
+                --library-install-path="/swift/usr/lib/swift/linux" \\
+                --module-install-path="/swift/usr/lib/swift/linux/x86_64"
+            """))
     subparsers = parser.add_subparsers(
-        description="Use one of these to specify whether to build, test, "
-                    "or install XCTest. If you don't specify any of these, "
-                    "'build' is executed as a default. You may also use "
-                    "'build' to also test and install the built products. "
-                    "Pass the -h or --help option to any of the subcommands "
-                    "for more information.")
+        description=textwrap.dedent("""
+            Use one of these to specify whether to build, test, or install
+            XCTest. If you don't specify any of these, 'build' is executed as a
+            default. You may also use 'build' to also test and install the
+            built products. Pass the -h or --help option to any of the
+            subcommands for more information."""))
 
     build_parser = subparsers.add_parser(
         "build",
-        description="Build XCTest.so, XCTest.swiftmodule, and XCTest.swiftdoc "
-                    "using the given Swift compiler. This command may also "
-                    "test and install the built products.")
+        description=textwrap.dedent("""
+            Build XCTest.so, XCTest.swiftmodule, and XCTest.swiftdoc using the
+            given Swift compiler. This command may also test and install the
+            built products."""))
     build_parser.set_defaults(func=_build)
     build_parser.add_argument(
         "--swiftc",
         help="Path to the 'swiftc' compiler that will be used to build "
              "XCTest.so, XCTest.swiftmodule, and XCTest.swiftdoc. This will "
              "also be used to build the tests for those built products if the "
              "--test option is specified.",
-        metavar="PATH",
```

Comment:
Thanks for the explanation! Looks great. 

I'm pretty sure I'd trust your judgement over mine on things Python 😆 Always looking to learn.